projects/core/src/asm/http-interceptors/csagent-auth-error.interceptor.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest,
+} from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, of, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import {
+  InterceptorUtil,
+  USE_CUSTOMER_SUPPORT_AGENT_TOKEN,
+} from '../../occ/utils/interceptor-util';
+import { CustomerSupportAgentErrorHandlingService } from '../../asm/services/csagent-error-handling.service';
+
+@Injectable({ providedIn: 'root' })
+export class CustomerSupportAgentAuthErrorInterceptor
+  implements HttpInterceptor {
+  constructor(
+    private csagentErrorHandlingService: CustomerSupportAgentErrorHandlingService
+  ) {}
+
+  intercept(
+    request: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    const isCustomerSupportAgentRequest = this.isCustomerSupportAgentRequest(
+      request
+    );
+    if (isCustomerSupportAgentRequest) {
+      request = InterceptorUtil.removeHeader(
+        USE_CUSTOMER_SUPPORT_AGENT_TOKEN,
+        request
+      );
+    }
+
+    return next.handle(request).pipe(
```

Comment:
Now that there are 2 different interceptors handling the 401 error, do we have a way to make sure, in the case of a customer support request 401 error that the other interceptor doesn't handle the 401 before this one? (this preventing this interceptor from running)